controllers/imageclusterinstall_controller.go

```diff
@@ -207,21 +210,32 @@ func (r *ImageClusterInstallReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	// in case there is no bmh we should set requirements met condition to true with image ready message
 	// in case bmh was set we will set this condition after host validations
-	if ici.Status.BareMetalHostRef == nil {
+	if ici.Spec.BareMetalHostRef == nil {
```

Comment:
I guess this was a typo originally? Not sure why else this would be changing here.

Reply:
Yeah, i believe we should check Spec and not status here